python/ray/air/config.py

```diff
@@ -756,6 +766,51 @@ def __post_init__(self):
         if not self.checkpoint_config:
             self.checkpoint_config = CheckpointConfig()
 
+        # Convert Paths to strings
+        if isinstance(self.local_dir, Path):
+            self.local_dir = str(self.local_dir)
+
+        if isinstance(self.storage_path, Path):
+            self.storage_path = str(self.storage_path)
+
+        local_path, remote_path = _resolve_storage_path(
+            self.storage_path, self.local_dir, self.sync_config.upload_dir
+        )
+
+        if self.sync_config.upload_dir:
+            assert remote_path == self.sync_config.upload_dir
+            warnings.warn(
+                "Setting a `SyncConfig.upload_dir` is deprecated and will be removed "
+                "in the future. Pass `RunConfig.storage_path` instead."
+            )
+            # Set upload_dir to None to avoid further downstream resolution.
+            # Copy object first to not alter user input.
+            self.sync_config = copy.copy(self.sync_config)
+            self.sync_config.upload_dir = None
+
+        if self.local_dir:
+            assert local_path == self.local_dir
+            warnings.warn(
+                "Setting a `RunConfig.local_dir` is deprecated and will be removed "
+                "in the future. Set `RunConfig.storage_path` instead or set the"
```

Comment:
These two have different meanings right?
One is for *ultimate* storage path and the other one is cached location and more for advanced users?
This sentence somehow makes me think that they have similar semantics.

Reply:
I've updated the comment, hope this makes it clearer!